internal/runners/platforms/platforms.go

```diff
@@ -71,9 +75,77 @@ func prepareParams(ps Params) (Params, error) {
 		ps.BitWidth = 32 << (^uint(0) >> 63) // gets host word size
 	}
 
+	var err error
 	if ps.Version == "" {
-		return ps, errors.New(locale.T("err_bad_platform_version"))
+		ps.Version, err = latestPlatformVersion(ps.Name)
+		if err != nil {
+			return ps, locale.WrapError(err, "err_prepare_params_latest", "Could not determine latest version for the platform: {{.V0}}", ps.Name)
+		}
 	}
 
 	return ps, nil
 }
+
+func splitNameAndVersion(input string) (string, string) {
+	nameArg := strings.Split(input, "@")
+	name := nameArg[0]
+	version := ""
+	if len(nameArg) == 2 {
+		version = nameArg[1]
+	}
+
+	return name, version
+}
+
+// platformSemver covers the semvers used by the platform that do not conform
+// to the semver standard. (ie. they have an extra numerical value after patch)
+type platformSemver struct {
+	semver     semver.Version
+	additional int
+}
+
+func latestPlatformVersion(name string) (string, error) {
```

Comment:
This is doing too much. This type of logic belongs on the solver. For now if they don't supply a version just use the constants here:

http://github.com/ActiveState/cli/blob/c91da98bb5a4f39f6823f713ac381d68b3581745/internal/constants/constants.go#L184-L184